scripts/inject-deterministic.py

```diff
@@ -4,14 +4,129 @@
+        """
+         Given a nonce, getScriptWithNonce returns the injected script text with the nonce.
+        If nonce is an empty string, getScriptWithNonce returns the script block
+        without attaching a nonce attribute.
+        Some responses may specify a nonce inside their Content-Security-Policy,
+        script-src directive.
+        The script injector needs to set the injected script's nonce attribute to
+        open execute permission for the injected script.
+        """
+
+        script_with_nouce = "<script"
+
+        if not nonce == "":
+            script_with_nouce += ' nonce="%s"' % nonce
+
+        script_with_nouce += ">"
+        script_with_nouce += script
+        script_with_nouce += "</script>"
+
+        return script_with_nouce
+
+    def get_nonce_from_headers(self, header):
+        """
+        get_nonce_from_headers returns the nonce token from a
+        Content-Security-Policy (CSP) header's script source directive, or an empty
+        string if the CSP header's script source does not contain a nonce.
+        For more background information on CSP and nonce, please refer to
+        https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/
+        Content-Security-Policy/script-src
+        https://developers.google.com/web/fundamentals/security/csp/
+        """
+
+        csp_script_directives = self.get_csp_script_directives(header)
+        if not csp_script_directives == "":
+            tokens = csp_script_directives.split(" ")
+            for token in tokens:
+                if token.startswith("'nonce-"):
+                    token = token.lstrip("'nonce-")
+                    token = token.rstrip("'")
+                    return token
+        return ""
+
+    def get_csp_script_directives(self, headers):
+        # get CSP script-src directives
+        if "Content-Security-Policy" in headers:
+            ctx.log.info("Original Header %s" % headers["Content-Security-Policy"])
+            csp = headers["Content-Security-Policy"]
+            directives = csp.split(";")
+
+            for directive in directives:
+                if directive.startswith("script-src "):
+                    return directive
+
+        return ""
+
+    def get_csp_header(self, headers, sha256):
+        # get_csp_header transforms a Content-Security-Policy (CSP) header to
+        # permit execution of inline scripts. Without this permission a page with a
+        # restrictive CSP will not execute injected scripts.
+        # For more background information on CSP, please refer to
+        # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/
+        # Content-Security-Policy/script-src
+        # https://developers.google.com/web/fundamentals/security/csp/
+
+        header_updated = False
+        if "Content-Security-Policy" in headers:
+            csp = headers["Content-Security-Policy"]
+            csp_directives = csp.split(";")
+        else:
+            return headers
```

Comment:
Rather than duplicate this logic, let's just call `get_csp_directives` from my previous comment.